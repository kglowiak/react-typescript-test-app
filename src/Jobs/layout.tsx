import React, { Component, Fragment } from 'react';

import { JobItem } from './JobItem/index';
import { JobsProps } from './index';

export class JobsLayout extends Component<JobsProps> {
  componentDidMount(): void {
    this.props.fetchJobs();
  }

  render() {
    const { jobs, loading } = this.props;
    return (
      <Fragment>
        <h1>HN Jobs</h1>
        {loading ? (
          <p>Loading...</p>
        ) : (
          jobs.map(job => <JobItem job={job} key={job.id} />)
        )}
      </Fragment>
    );
  }
}
