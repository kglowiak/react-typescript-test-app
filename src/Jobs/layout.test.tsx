import React from 'react';
import {render, shallow} from 'enzyme';
import { JobsProps } from './index';
import { JobsLayout } from './layout';

describe('JobsLayout', () => {
  const props: JobsProps = {
    fetchJobs: jest.fn(),
    jobs: [
      {
        type: 'job',
        time: 123455,
        score: 1,
        by: 'test',
        id: 123455,
        title: 'test 1',
        url: 'testurl',
      },
    ],
    loading: false,
  };

  it('should call fetchJobs once on component mount', () => {
    const wrapper = shallow(<JobsLayout {...props} />);
    expect(props.fetchJobs).toHaveBeenCalledTimes(1);
  });

  it('should render a component in loading state', () => {
    const wrapper = render(<JobsLayout {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a component in loading state', () => {
    const wrapper = render(<JobsLayout {...props} loading={false}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
