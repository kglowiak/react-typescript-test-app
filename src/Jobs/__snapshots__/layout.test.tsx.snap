// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsLayout should render a component in loading state 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "HN Jobs",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h1",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "class": "css-10mb6uy",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "class": "css-1m1p7ie",
            "href": "testurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "data": "test 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "css-uahybi",
            },
            "children": Array [
              Object {
                "data": "Added by test at 02/01/1970",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        Object {
          "attribs": Object {
            "class": "css-uahybi",
          },
          "children": Array [
            Object {
              "data": "Added by test at 02/01/1970",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "css-1m1p7ie",
              "href": "testurl",
              "target": "_blank",
            },
            "children": Array [
              Object {
                "data": "test 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
              "target": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
              "target": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "class": "css-10mb6uy",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-1m1p7ie",
                "href": "testurl",
                "target": "_blank",
              },
              "children": Array [
                Object {
                  "data": "test 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "css-uahybi",
                },
                "children": Array [
                  Object {
                    "data": "Added by test at 02/01/1970",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
                "target": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
                "target": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "css-uahybi",
              },
              "children": Array [
                Object {
                  "data": "Added by test at 02/01/1970",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-1m1p7ie",
                  "href": "testurl",
                  "target": "_blank",
                },
                "children": Array [
                  Object {
                    "data": "test 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                  "target": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                  "target": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "1": Object {
    "attribs": Object {
      "class": "css-10mb6uy",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1m1p7ie",
          "href": "testurl",
          "target": "_blank",
        },
        "children": Array [
          Object {
            "data": "test 1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-uahybi",
          },
          "children": Array [
            Object {
              "data": "Added by test at 02/01/1970",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-uahybi",
        },
        "children": Array [
          Object {
            "data": "Added by test at 02/01/1970",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-1m1p7ie",
            "href": "testurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "data": "test 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "data": "HN Jobs",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "h1",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "HN Jobs",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`JobsLayout should render a component in loading state 2`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "HN Jobs",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h1",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "class": "css-10mb6uy",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "class": "css-1m1p7ie",
            "href": "testurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "data": "test 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "css-uahybi",
            },
            "children": Array [
              Object {
                "data": "Added by test at 02/01/1970",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        Object {
          "attribs": Object {
            "class": "css-uahybi",
          },
          "children": Array [
            Object {
              "data": "Added by test at 02/01/1970",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "css-1m1p7ie",
              "href": "testurl",
              "target": "_blank",
            },
            "children": Array [
              Object {
                "data": "test 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
              "target": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
              "target": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "class": "css-10mb6uy",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-1m1p7ie",
                "href": "testurl",
                "target": "_blank",
              },
              "children": Array [
                Object {
                  "data": "test 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "css-uahybi",
                },
                "children": Array [
                  Object {
                    "data": "Added by test at 02/01/1970",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
                "target": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
                "target": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "css-uahybi",
              },
              "children": Array [
                Object {
                  "data": "Added by test at 02/01/1970",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-1m1p7ie",
                  "href": "testurl",
                  "target": "_blank",
                },
                "children": Array [
                  Object {
                    "data": "test 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                  "target": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                  "target": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "1": Object {
    "attribs": Object {
      "class": "css-10mb6uy",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1m1p7ie",
          "href": "testurl",
          "target": "_blank",
        },
        "children": Array [
          Object {
            "data": "test 1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-uahybi",
          },
          "children": Array [
            Object {
              "data": "Added by test at 02/01/1970",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-uahybi",
        },
        "children": Array [
          Object {
            "data": "Added by test at 02/01/1970",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-1m1p7ie",
            "href": "testurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "data": "test 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "data": "HN Jobs",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "h1",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "HN Jobs",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
